File: phplib/myview.php
<?php 
/* Here lies a simple template engine for PHP
 * 
 */
class MyView {
	protected $vars = array();
	public function __construct($template_dir = null) {
		$this->template_dir =  dirname(__FILE__).'/../templates/';
	}
	public function render($template_file) {
		if (! file_exists($this->template_dir.$template_file)) {
			throw new Exception('no template file ' . $template_file .
					' present in directory ' . $this->template_dir);
		}
		$tpl = file_get_contents($this->template_dir.$template_file);
		reset($this->vars);
		while( list($key, $val) = each($this->vars)){
			$tpl = str_replace("<!--$key-->", $val, $tpl);
		}
		return $tpl;
	}
	public function __set($name, $value) {
		$this->vars[$name] = $value;
	}
	public function __get($name) {
		return $this->vars[$name];
	}
	
}
?>

File: pylib/myview.py
"""Simple templating"""
import os


class MyView(object):

    def __init__(self):
        self.vars = dict()

    def render(self, template):
        fn = "%s/%s" % (os.path.join(os.path.dirname(__file__),
                        "../templates"), template)
        if not os.path.isfile(fn):
            raise Exception("template %s not found!" % (fn,))
        tpl = open(fn).read()
        for key, val in self.vars.iteritems():
            tpl = tpl.replace("<!--%s-->" % (key,), val)
        return tpl
